compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -2016,6 +2022,61 @@ fn add_order_independent_options(
     add_rpath_args(cmd, sess, codegen_results, out_filename);
 }
 
+// Write the debugger visualizer files for each crate to the target directory and gather the file paths.
+fn collect_debugger_visualizers(
+    tmpdir: &Path,
+    sess: &Session,
+    codegen_results: &CodegenResults,
+    debugger_visualizer_paths: &mut Vec<PathBuf>,
+) {
+    let local_crate_name = &codegen_results.crate_info.local_crate_name.to_string();
+    let natvis_str = "natvis";
+
+    for (cnum, debugger_visualizers) in codegen_results.crate_info.debugger_visualizers.clone() {
+        for debugger_visualizer in debugger_visualizers {
+            let crate_name = if cnum == LOCAL_CRATE {
+                local_crate_name
+            } else {
+                &codegen_results.crate_info.crate_name[&cnum]
+            };
+
+            let mut hasher = DefaultHasher::new();
+            hasher.write(&debugger_visualizer.src_hash.hash_bytes());
+            let hash = hasher.finish();
+
+            let visualizer_ext = match debugger_visualizer.visualizer_type {
+                DebuggerVisualizerType::Natvis => natvis_str,
+            };
+
+            let visualizer_out_file =
+                tmpdir.join(format!("{}-{}.{}", crate_name, hash, visualizer_ext));
```

Comment:
Do we really need the source hash? We could just generate simple indices from a counter here.

Also note, the crate name does not uniquely identify a crate within the crate graph. There could be multiple versions of the same crate (and they all could have the exact same NatVis files in them, so the src_hash would not disambiguate them either).

I think it might make sense to just generate filenames like `dbgvis-<index>.natvis` here. Or maybe `dbgvis-<crate-name>-<index>.natvis` for increased debuggability. But in both cases `<index>` would come from a single counter, so that no two names generated in this function are the same.

Reply:
Removed the source hash as well. Now using a counter to ensure the filenames stay unique.